src/conoffsetcollect/kafka_offset_collect.go

```diff
@@ -340,13 +340,25 @@ func populateOffsetStructs(offsets, hwms groupOffsets) []*partitionOffsets {
 
 	var poffsets []*partitionOffsets
 	for topic, partitions := range hwms {
+		topicOffsets, ok := offsets[topic]
+		if !ok {
+			log.Error("Offsets not collected for topic %s")
+			continue
+		}
+
 		for partition, hwm := range partitions {
+			offset, ok := topicOffsets[partition]
+			if !ok {
+				log.Error("Offset not collected for topic %s, partition %d", topic, partition)
+				continue
```

Comment:
We should still report the HighWaterMark if possible and just not the other two metrics.